### PR TITLE
feat(issue-priority): Set the priority for new events

### LINKED ISSUE #63310: Modify group creation pathway to set priority for new issues

### PR DESCRIPTION
**Pull request by snigdhas**
Set the priority for new issues using the level and first event severity.

Fixes https://github.com/getsentry/sentry/issues/63310

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -2049,6 +2055,34 @@ def _get_severity_metadata_for_group(event: Event) -> Mapping[str, Any]:
...
+    elif level == logging.FATAL:
+        return PriorityLevel.HIGH
+
+    elif level == logging.WARNING:
+        if not severity_score or severity_score < 0.1:
+            return PriorityLevel.MEDIUM
+
+        return PriorityLevel.HIGH  # severity_score >= 0.1
+    elif level == logging.ERROR:
+        if not severity_score or severity_score >= 0.1:
```
Comment (malwilley) [inline src/sentry/event_manager.py:2077]: I think you want `severity_score is None` in case the value is `0.0`

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -1811,7 +1811,13 @@ def _create_group(project: Project, event: Event, **kwargs: Any) -> Group:
     group_data.setdefault("metadata", {}).update(sdk_metadata_from_event(event))
 
     # add severity to metadata for alert filtering
-    group_data["metadata"].update(_get_severity_metadata_for_group(event))
+    severity = _get_severity_metadata_for_group(event)
+    group_data["metadata"].update(severity)
+
+    if features.has("projects:issue-priority", project, actor=None):
+        priority = _get_priority_for_group(severity, kwargs)
+        kwargs.update({"priority": priority})
```
Comment (wedamija) [inline src/sentry/event_manager.py:1819]: Nit: `kwargs["priority"] = priority`

File: src/sentry/event_manager.py [hunk 3]
```diff
@@ -1811,7 +1811,13 @@ def _create_group(project: Project, event: Event, **kwargs: Any) -> Group:
...
+        group_data["metadata"].update({"initial_priority": priority})
```
Comment (wedamija) [inline src/sentry/event_manager.py:1820]: Nit: `group_data["metadata"]["initial_priority"] = priority`

File: src/sentry/event_manager.py [hunk 4]
```diff
@@ -2049,6 +2055,34 @@ def _get_severity_metadata_for_group(event: Event) -> Mapping[str, Any]:
...
+    severity_score = severity.get("severity", None)
+
+    if level in [logging.INFO, logging.DEBUG]:
+        return PriorityLevel.LOW
+
+    elif level == logging.FATAL:
+        return PriorityLevel.HIGH
+
+    elif level == logging.WARNING:
+        if not severity_score or severity_score < 0.1:
```
Comment (wedamija) [inline src/sentry/event_manager.py:2072]: Feels like maybe the 0.1 should be a constant, `HIGH_PRIORITY_THRESHOLD` or something?

File: src/sentry/event_manager.py [hunk 5]
```diff
@@ -2049,6 +2057,34 @@ def _get_severity_metadata_for_group(event: Event) -> Mapping[str, Any]:
...
+
+        return PriorityLevel.HIGH  # severity_score >= HIGH_SEVERITY_THRESHOLD
+    elif level == logging.ERROR:
+        if severity_score is None or severity_score >= HIGH_SEVERITY_THRESHOLD:
+            return PriorityLevel.HIGH
+
+        return PriorityLevel.MEDIUM  # severity_score < HIGH_SEVERITY_THRESHOLD
+
+    logger.warning("unknown log level %s or severity score %s", level, severity_score)
+    return PriorityLevel.MEDIUM
```
Comment (wedamija) [inline src/sentry/event_manager.py:2085]: Just being overcautious but let's wrap this with a try/catch just to be safe

**Review by wedamija — APPROVED** [review]

**Pull request merged**